index.bs

```diff
@@ -90,6 +100,37 @@ The exact type of the `globalScope` can vary across runtimes. Most Web Platform
 
 With many runtimes, adding a new global-scoped property can introduce breaking changes when the new global conflicts with existing application code. Many Web Platform APIs define global properties using the `readonly` attribute. To avoid introducing breaking changes, runtimes conforming to this specification MAY choose to ignore the `readonly` attribute for properties being added to the global scope.
 
+Requirements for EcmaScript modules {#esm-requirements}
+=======================================================
+
+[=Web-interoperable runtimes=] which support EcmaScript modules must implement the following ECMA-262 host hooks as follows:
+
+: [$HostLoadImportedModule$]
+:: This host hook must be implemented [[HTML#hostloadimportedmodule|as defined in the HTML spec]], except that they may pass a [=perform the fetch hook=] algorithm to [=fetch a single imported module script=] even when <var ignore="">loadState</var> is undefined. (See also [[#custom-module-loading]].)
+: [$HostGetSupportedImportAttributes$]
+:: This host hook must be implemented [[HTML#hostgetsupportedimportattributes|as defined in the HTML spec]].
+: [$HostGetImportMetaProperties$]
+:: If a runtime can never have an [=entrypoint module=] for any [=agent clusters=] (e.g. web browsers), then it may implement this host hook [[HTML#hostgetimportmetaproperties|as defined in the HTML spec]]. Otherwise, it must implement it as follows:
+     1. Let <var>moduleScript</var> be <var ignore="">moduleRecord</var>.\[[HostDefined]].
+     1. Let <var>urlString</var> and <var>resolveFunction</var> be [[HTML#hostgetimportmetaproperties|as defined in the HTML spec]].
+     1. Let <var>is main</var> be true if <var>moduleScript</var> is the [=surrounding agent=]'s [=agent cluster=]'s [=entrypoint module=]; false otherwise.
+     1. Return « [=Record=] { \[[Key]]: "<code>url</code>", \[[Value]]: <var>urlString</var> }, [=Record=] { \[[Key]]: "<code>resolve</code>", \[[Value]]: <var>resolveFunction</var> }, [=Record=] { \[[Key]]: "<code>main</code>", \[[Value]]: <var>is main</var> } ».
```

Comment:
Do we want to define more strictly what the `resolveFunction` implementation should look like? Like how many arguments does it take, and what its return value should look like. Asking because I think Node.js would like to have an optional second parameter to allow access to the resolve value from another module's perspective. Also it'd be convenient for us if the return value could be a `URL` instance.

Reply:
See https://github.com/nodejs/modules/issues/550 and https://github.com/whatwg/html/pull/5572. Node has been engaging with WHATWG for years on this, and at this point I think we’ve accepted that we’re following whatever signature is allowed by the HTML spec. I would be nervous if WinterCG approved an API that differed from HTML’s.

We could perhaps introduce some new API, `import.meta.resolveURL` say, for a different variation.